babel-plugins/postfix-if.js

```diff
@@ -0,0 +1,163 @@
+    var inspector = inspectorForPath(path);
+    return inspector.outerPath(path);
+  }
+
+  // Raw consequent (not necessarily a statement)
+  var getConsequent = function (path) {
+    var inspector = inspectorForPath(path);
+    return inspector.consequent(path);
+  }
+
+  var grandparentPath = function (path) {
+    return path.parentPath.parentPath;
+  }
+
+  var assertValidSequence = function (node) {
+    var err = function () {
+      throw new Error('Postfix if/unless must be last expression in a two-expression sequence');
+    }
+    if (node.expressions.length !== 2) {
+      err();
+    }
+
+    var firstExpression = node.expressions[0];
+    if (t.isCallExpression(firstExpression) && calleePattern.test(firstExpression.callee.name)) {
+      err();
+    }
+  }
+
+  // Get the inspector which will answer questions about the path
+  var inspectorForPath = function (path) {
+    var inspector = nodeInspectors.find(inspector => inspector.test(path));
+    if (!inspector) throw new Error('No inspector matched the node');
+    return inspector;
+  }
+
+  var bareReturnInspector = {
+    test: function (path) {
+      return path.parent.type === 'ReturnStatement';
+    },
+
+    consequent: function (path) {
+      return t.returnStatement();
+    },
+
+    outerPath: function (path) {
+      return path.parentPath;
+    },
+  };
+
+  var unaryCompletionInspector = {
+    test: function (path) {
+      var sequence = path.parent;
+      if (!t.isSequenceExpression(sequence)) return false;
+
+      assertValidSequence(sequence);
+
+      // Grandparent must be return or throw statement
+      var grandparent = grandparentPath(path).node;
+      var grandparentPattern = /^(Return|Throw)Statement$/;
+      if (!grandparentPattern.test(grandparent.type)) return false;
+
+      return true;
+    },
+
+    consequent: function (path) {
+      var grandparent = grandparentPath(path);
+      var sequence = path.parent;
+      var argument = sequence.expressions[0];
+      return t[grandparent.node.type](argument);
+    },
+
+    outerPath: function (path) {
+      return grandparentPath(path);
+    },
+  };
+
+  var expressionInspector = {
+    test: function (path) {
+      var sequence = path.parent;
+      if (!t.isSequenceExpression(sequence)) return false;
+      assertValidSequence(sequence);
+
+      if (!t.isExpressionStatement(grandparentPath(path).node)) return false;
+      return true;
+    },
+
+    consequent: function (path) {
+      var sequence = path.parent;
+      return sequence.expressions[0];
+    },
+
+    outerPath: function (path) {
+      return grandparentPath(path);
+    },
+  };
+
+  var nodeInspectors = [
+    bareReturnInspector,
+    unaryCompletionInspector,
+    expressionInspector,
+  ];
+
+  return {
+    visitor: {
+      CallExpression: function (path) {
+        var callee = path.node.callee;
+
+        if (callee.type !== 'Identifier') return;
+        if (!calleePattern.test(callee.name)) return;
```

Comment:
isn't this expected to be exactly either `If` or `Unless` would seem better to me to not use the regex, but just an array to compare with